client/state/sites/reducer.js

```diff
@@ -214,6 +226,32 @@ export function items( state = {}, action ) {
 
 			return state;
 		}
+
+		case InvitesActionTypes.RECEIVE_INVITE_ACCEPTED_SUCCESS: {
+			const { invite, data } = action;
+
+			if ( ! invite || ! data || ! data.sites ) {
+				return state;
+			}
+
+			if ( invite.role === 'follower' || invite.role === 'viewer' ) {
+				return state;
+			}
+
+			const siteId = get( invite, 'site.ID', null );
+
+			if ( ! siteId || state[ siteId ] ) {
+				return state;
+			}
+
+			const site = find( data.sites, receivedSite => receivedSite.ID === siteId );
```

Comment:
The `data.sites` array contains the complete updated list of sites. You should do the equivalent of `SITES_RECEIVE` here -- update the whole list. Correct me if I'm wrong.

Reply:
Do we really want to update all the data just for adding one more site? I know we do the same in `sites-list` but it seemed pretty non-performant. I intentionally did it like this. Or do you think there might be some corner cases we are not aware of?

Reply:
The current code always reloads the whole sites list when a new site is added. It happens when accepting an invite, when the REST API call itself returns the new sites list as result. It also happens after connecting a new Jetpack site in `client/state/jetpack-connect/actions.js`, where a `wpcom.me().sites()` request is issued after successful Jetpack authorization.

So I'm wondering if it has some reason. Maybe some attributes of other sites get updated, too? The first one that comes to mind is `site.visible`: is the list of visible sites changed after adding a new one?

I'd recommend to keep the current behavior of updating the whole list -- we'll have to investigate closely what exactly happens on the backend before changing it.

Reply:
Alright. Even if it was an oversight, we could improve later. This task is about reduxification of current code.

Reply:
Fixed in dd5b4d1.